lfs-image

```diff
@@ -14,24 +14,44 @@ fi
 
 export BUILD_DATE
 BUILD_DATE="$(date "+%Y%m%d-%H%M")"
-LUA_FILES=$(find ../lua -iname "*.lua")
+
+if [ -z "$1" ]; then
+  cd /opt/lua
```

Comment:
That's hard to do here, as the following commands inside the if rely on it.

Reply:
True, got it. The problem is that at the end of `if/else` the current directory is different depending on which branch was active. Then further down in `make_image ` you make assumptions about the current directory for `find`. That is hard to trace.

Idea:
- small function `cd_firmware_base` or so
- call it on line 13 and at the end of this `if/else`
- rather than using a relative `find ../lua` you use `${lua_base}`

What did I miss?

Reply:
didn't see the ../lua. fixed that.
But the path set in the if/else is also needed in line 50. So every path is absolute now except the path where the find is executed or the LUA_FILES are relative to.
Setting the path back and forth does not make the script more readable IMHO.
since all other pathes are absolute it does not really matter for the rest of the script what folder it is in.

Reply:
👍 for fixed paths. Guess you still need to push tho'.